gateway/src/main.rs

```diff
@@ -237,46 +254,93 @@ async fn init(db: SqlitePool, args: InitArgs) -> io::Result<()> {
             );
+            let certs = create_gateway_certificate(fs, public, acme, resolver).await;
+            certs.clone().save_pem(&tls_path).unwrap();
+            certs
+        }
+    }
+}
 
-            if !creds_path.exists() {
-                panic!(
-                    "no ACME credentials found at {}, cannot continue with certificate creation",
-                    creds_path.display()
-                );
-            }
-
-            let creds = std::fs::File::open(creds_path).unwrap();
-            let creds: AccountCredentials = serde_json::from_reader(&creds).unwrap();
-
-            let identifier = format!("*.{public}");
-
-            // Use ::Dns01 challenge because that's the only supported
-            // challenge type for wildcard domains
-            let (chain, private_key) = acme
-                .create_certificate(&identifier, ChallengeType::Dns01, creds)
-                .await
-                .unwrap();
-
-            let mut buf = Vec::new();
-            buf.extend(chain.as_bytes());
-            buf.extend(private_key.as_bytes());
+async fn create_gateway_certificate<P: AsRef<Path>>(
+    fs: P,
+    public: FQDN,
+    acme: AcmeClient,
+    resolver: Arc<GatewayCertResolver>,
+) -> ChainAndPrivateKey {
+    let creds_path = fs.as_ref().join("acme.json");
+    let identifier = format!("*.{public}");
+    if !creds_path.exists() {
+        panic!(
+            "no ACME credentials found at {}, cannot continue with certificate creation",
+            creds_path.display()
+        );
+    }
+    let creds = std::fs::File::open(creds_path).expect("Invalid credentials path");
+    let creds: AccountCredentials =
+        serde_json::from_reader(&creds).expect("Can not parse admin credentials from path");
+    let account = Account::from_credentials(creds)
+        .map_err(|error| {
+            error!(
+                error = &error as &dyn std::error::Error,
+                "failed to convert acme credentials into account"
+            );
+            AcmeClientError::AccountCreation
+        })
+        .unwrap();
 
-            let certs = ChainAndPrivateKey::parse_pem(Cursor::new(buf)).unwrap();
+    // Use ::Dns01 challenge because that's the only supported
+    // challenge type for wildcard domains
+    let (chain, private_key) = acme
+        .create_certificate(&identifier, ChallengeType::Dns01, &account)
+        .await
+        .unwrap();
 
-            certs.clone().save_pem(&tls_path).unwrap();
+    let mut buf = Vec::new();
+    buf.extend(chain.as_bytes());
+    buf.extend(private_key.as_bytes());
+    let certs = ChainAndPrivateKey::parse_pem(Cursor::new(buf)).unwrap();
+    resolver.serve_default_der(certs.clone()).await.unwrap();
+    certs
+}
 
-            certs
+async fn automate_renewal<P: AsRef<Path> + Clone>(
+    fs: P,
+    public: FQDN,
+    acme: AcmeClient,
+    resolver: Arc<GatewayCertResolver>,
+) {
+    loop {
+        tokio::time::sleep(Duration::from_secs(240)).await;
```

Comment:
Nit: Why not sleep/wait until the first time when we actually want this to fire? Instead of `loop { ... }`?

Reply:
I think this will not be required anymore given we can not properly automate the gateway certificate renewal without the TXT record insertion for the DNS challenge. I can open an issue to track this issue and handle it separately, is that ok?